geoips/geoips_utils.py

```diff
@@ -445,3 +449,129 @@ def merge_nested_dicts(dest, src, in_place=True):
+                log.interactive("-" * len(f"Available {pkg_name.title()} Commands"))
+                table_data = []
+                for name, cmd in scripts.items():
+                    table_data.append([name, cmd])
+                log.interactive(
+                    tabulate(
+                        table_data,
+                        headers=["Command Name", "Command Path"],
+                        tablefmt="rounded_grid",
+                        maxcolwidths=get_terminal_size().columns // 2,
+                    )
+                )
+            # Otherwise let the user know that there were not commands found for this
+            # package.
+            else:
+                log.interactive(f"No '{pkg_name.title()}' Commands were found.")
+
+    except FileNotFoundError:
+        log.interactive(
+            f"No pyproject.toml file found at {toml_path}. Please create one to move "
+            "forward."
+        )
+
+
+def _get_pyproj_scripts(pyproj):
+    """Get command scripts out of the provided pyproject.toml file.
+
+    Parameters
+    ----------
+    pyproj: dict
+        - The pyproject.toml file represented as a dictionary.
+    """
+    try:
+        # Try to access the scripts via poetry's representation
+        scripts = pyproj["tool"]["poetry"]["scripts"]
+    except KeyError:
+        # If that didn't work, they don't exist or it's setuptools-based.
+        # Try to access scripts via setuptools representation.
+        try:
+            scripts = pyproj["project"]["entry-points"]["console_scripts"]
+        except KeyError:
+            # Scripts were not specified, return None
+            scripts = None
+    return scripts
+
+
+def _get_pkg_name_and_logger(pkg_name, provided_log):
+    """Retrieve the Package Name and Logger from the provided variables.
+
+    If pkg_name is None, retrieve pkg_name from the commandline arguments
+    (defaults to 'geoips'). If log is None, set log to the LOG attribute found in this
+    module.
+
+    Parameters
+    ----------
+    pkg_name: str
+        - The name of the GeoIPS Plugin Package whose command's will be exposed.
+        - If None, assume this was called via the commandline and retrieve package_name
+          via that manner. Otherwise use the supplied package_name.
+    provided_log: logging.Logger
+        - If not None, use this log so we can test the output of this function for
+          testing purposes, otherwise retrieve the LOG attribute from this module.
+
+    Returns
+    -------
+    pkg_name, log: str, logging.Logger
+        - The name of the package to retrieve commands from and the logger used to
+          output it.
+    """
+    plugin_packages = [
+        str(ep.value) for ep in get_entry_point_group("geoips.plugin_packages")
+    ]
+    if provided_log:
+        log = provided_log
+    else:
+        log = LOG
+    if pkg_name is None:
+        # This function was called via the command line.
```

Comment:
Or someone passed `None`

Reply:
Fair. I don't see why that would ever happen, but I'll note that here nonetheless. Best to document all cases.

Reply:
Haha yep, you can never predict [what people will do](https://news.bme.com/wp-content/uploads/2008/01/code-tattoo.jpg) with your code.

Reply:
Updated this to say providing None is possible as well.